allennlp/tests/training/learning_rate_schedulers_test.py

```diff
@@ -222,3 +223,114 @@ def init_and_restore_scheduler(optimizer: torch.optim.Optimizer,
 
             for it, lr in lr_checks:
                 assert lrs[it] == lr
+
+
+class SlantedTriangularTest(AllenNlpTestCase):
+
+    def setUp(self):
+        super(SlantedTriangularTest, self).setUp()
+        self.model = torch.nn.Sequential(OrderedDict([
+            ('lin1', torch.nn.Linear(10, 10)),
+            ('lin2', torch.nn.Linear(10, 10))
+        ]))
+
+    def _get_optimizer(self, lr: float = 1.0):
+        optimizer_params = Params({"type": "sgd", "lr": lr})
+        optimizer_params["parameter_groups"] = [[[f"^{m}"], {}] for m in self.model._modules]
+        return Optimizer.from_params(self.model.named_parameters(), optimizer_params)
+
+    def test_from_params(self):
+        optim = self._get_optimizer()
+        sched = LearningRateScheduler.from_params(
+            optim, Params({"type": "slanted_triangular",
+                           "num_epochs": 5,
+                           "num_steps_per_epoch": 10,
+                           "gradual_unfreezing": True,
+                           "discriminative_fine_tuning": True,
+                           "decay_factor": 0.5})).lr_scheduler
+
+        assert sched.num_epochs == 5
+        assert sched.num_steps_per_epoch == 10
+        assert sched.gradual_unfreezing is True
+        assert sched.freezing_current is True
+
+        assert len(optim.param_groups) == 3
+        # The default parameter group in the Optimizer is empty
+        assert not optim.param_groups[-1]["params"]
+        assert optim.param_groups[-2]["lr"] == 1.0
+        assert optim.param_groups[-3]["lr"] == 0.5
+
+        with self.assertRaises(TypeError):
+            # num_epochs and num_steps_per_epoch are required
+            LearningRateScheduler.from_params(optim, Params({"type": "slanted_triangular", "num_epochs": 5}))
+            LearningRateScheduler.from_params(optim, Params({"type": "slanted_triangular", "num_steps_epochs": 10}))
+
+    def test_schedules(self):
+        slanted_triangular_cases: List[Tuple[Dict[str, Any], List[Tuple[int, float]]]] = [
+            (
+                    {"num_epochs": 5,
+                     "num_steps_per_epoch": 10,
+                     "gradual_unfreezing": True},  # parameters
+                    [(0, 1, 0.03125),  # iteration, layer, learning rate
```

Comment:
I think it's this spacing that pylint doesn't like.  Sorry, this error seems particularly annoying, because your spacing doesn't seem unreasonable.